### PR TITLE
Improved UI for dynamic variables

### LINKED ISSUE #611: Improve UI for dynamic variables
Part of https://github.com/bigbinary/neeto-engineering-ui-ux/issues/113

@praveen-murali-ind _a

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #611 

**Description**

- Changed: ui of variables from `Button` to `Tag`

**Checklist**

~~- [ ] I have made corresponding changes to the documentation.~~
- [x] I have added the necessary label (patch/minor/major - If package publish is required).

**Reviewers**
@AbhayVAshokan _a

### CODE REVIEW

File: lib/components/Editor/CustomExtensions/Variable/VariableList.jsx [hunk 1]
```diff
@@ -34,15 +35,15 @@ const VariableCategory = ({ index, title, variables, onClickItem }) => (
...
-        <button
-          className="neeto-editor-variable"
+        <Tag
           data-cy={`neeto-editor-variable-option-item-${item.label}--${item.value}`}
           key={`${item.label}--${item.value}`}
-          type="button"
+          label={item.label}
+          style="secondary"
+          className="neeto-ui-text-black
+                    neeto-ui-border-gray-300 hover:neeto-ui-text-primary-800 hover:neeto-ui-border-primary-800 neeto-ui-leading-snug cursor-pointer px-2 ne_variable-tag"
```
Comment (AbhayVAshokan) [inline lib/components/Editor/CustomExtensions/Variable/VariableList.jsx:44]: @praveen-murali-ind _a please review.

If I understand correctly, we need not override any of the default styles of the `Tag` component and let neetoUI handle the styling.

@gaagul please delete the class definition of unused styles (if any). In this case the `neeto-editor-variable` class is deleted.

**Comment by praveen-murali-ind** [comment]
@gaagul _a Looks good to me. A few changes.

- Let’s not override the default styles of neetoUI Tag ( The updated design for the variable item is slightly different than the neetoUI tag styles ). We can use a custom button component for this.
- Please left align the sub headings
- Use `color: rgb(var(--neeto-ui-black));` for the sub heading
- Update the sub heading margin bottom to `10px`

After these changes
Screenshot 2023-02-21 at 3 02 39 PM

Current UI

Screenshot 2023-02-21 at 2 49 17 PM

**Comment by gaagul** [comment]
@AbhayVAshokan _a Please verify, and let me know if style need to be improved.

**Review by AbhayVAshokan — APPROVED** [review]
@praveen-murali-ind _a

**Comment by praveen-murali-ind** [comment]
1. Please update the hover state of items to match the design

Existing UI

Screenshot 2023-02-23 at 11 45 56 AM

Design

image

---

2. Update the focus visible styles of items. Currently we are using yellow outline. Let’s use the same hover style mentioned above.

Screenshot 2023-02-23 at 10 46 04 AM

---

3. Add `user-select: none` to the custom buttons.

**Comment by praveen-murali-ind** [comment]
@gaagul _A Please work on the above changes. 

Ref for text selection - https://neeto-engineering.neetokb.com/articles/when-is-it-ok-to-disable-text-selection

**Comment by gaagul** [comment]
@AbhayVAshokan _a I have made the above mentioned changes.

**Comment by AbhayVAshokan** [comment]
@praveen-murali-ind _a

File: lib/styles/editor/_menu.scss [hunk 2]
```diff
@@ -147,18 +147,20 @@
...
-        transition: var(--neeto-ui-transition);
-
-        &:hover {
-          background-color: rgb(var(--neeto-ui-info-100));
-        }
+      button {
+        padding: 6px;
+        border: solid 1px rgb(var(--neeto-ui-gray-300));
+        border-radius: var(--neeto-ui-rounded-md);
+        user-select: none;
```
Comment (praveen-murali-ind) [inline lib/styles/editor/_menu.scss:154]: @gaagul _a Add `color: rgb(var(--neeto-ui-black));` to button.

File: lib/styles/editor/_menu.scss [hunk 3]
```diff
@@ -147,18 +147,20 @@
...
+      }
+
+      button:hover {
```
Comment (praveen-murali-ind) [inline lib/styles/editor/_menu.scss:157]: Combine the selectors for `button:hover`, `button:focus` and `button:focus-visible` and apply the following rules.

```
    button:focus-visible,
    button:focus,
    button:hover{
      color: rgb(var(--neeto-ui-primary-800));
      border-color: rgb(var(--neeto-ui-primary-800));
      outline: none;
      box-shadow: none;
    }
```

**Comment by gaagul** [comment]
@praveen-murali-ind _a I have made the changes.

**Comment by praveen-murali-ind** [comment]
@AbhayVAshokan @gaagul Looks good to me.

**Pull request merged**